prov/sockets/include/sock.h

```diff
@@ -413,13 +413,13 @@ enum {
  */
 struct sock_op {
 	uint8_t op;
-	uint8_t src_iov_len;
-	uint8_t	dest_iov_len;
+	uint64_t src_iov_len;
+	uint64_t dest_iov_len;
 	struct {
-		uint8_t	op;
-		uint8_t	datatype;
-		uint8_t	res_iov_len;
-		uint8_t	cmp_iov_len;
+		uint8_t		op;
+		uint8_t		datatype;
+		uint64_t	res_iov_len;
+		uint64_t	cmp_iov_len;
 	} atomic;
 	uint8_t	reserved[1];
 };
```

Comment:
This structure is part of the wire protocol.  We need to avoid changing it.

Reply:
@shefty , sure. there are a couple of similiar to that places. I'll fix all.
Should it be fixed for the `struct sock_msg_hdr`?

Reply:
We don't want to modify the wire protocol, so casts will be needed when filling out the wire structure fields.